include/zephyr/drivers/clock_control/nrf_clock_control.h

```diff
@@ -190,6 +190,8 @@ __subsystem struct nrf_clock_control_driver_api {
 	int (*cancel_or_release)(const struct device *dev,
 				 const struct nrf_clock_spec *spec,
 				 struct onoff_client *cli);
+	void (*request_zli)(const struct device *dev);
+	void (*release_zli)(const struct device *dev);
```

Comment:
I don't think we should extend this structure. These functions will be available only for HFXO, so why not just go with `nrf_clock_control_hfxo_request` and `nrf_clock_control_hfxo_release` (in the legacy clock control we had `z_nrf_clock_bt_ctlr_hf_request` and `z_nrf_clock_bt_ctlr_hf_release` to meet similar needs).

Reply:
that is true for now, until a new clock is needed, at which point the BLE stack has to be changed to use a different API. I don't know how portable the BLE stack is, but this would help if it used the devicetree

Reply:
As of now I changed it as @anangl  suggested. Following experience from nrf clock control there were no need for anything more than ZLI/BLE API. If there is another kind of HFCLK. Also I've left the device parameter so if there are multiple HFCLK devices the API is ready for it.